src/blocked.c

```diff
@@ -174,57 +198,157 @@ void queueClientForReprocessing(client *c) {
     }
 }
 
+/* Unblock a client which is currently Blocked on and provided a timeout.
+ * In this case since we might have a command pending
+ * we want to remove the pending flag to indicate we already responded to the
+ * command with timeout reply. */
+void unblockClientOnTimeout(client *c) {
+    if (c->flags & CLIENT_PENDING_COMMAND)
+        c->flags &= ~CLIENT_PENDING_COMMAND;
+    unblockClient(c);
+}
+
+/* Helper function to unblock a client that's waiting in a blocking operation such as BLPOP.
+ * You should never call this function directly, but unblockClient() instead. */
```

Comment:
```suggestion
 * Internal function for unblockClient(). */
```

i was wondering who's "You"?